app/src/main/java/org/schabi/newpipe/player/ui/MainPlayerUi.java

```diff
@@ -696,10 +698,24 @@ public void onScrolledDown(final RecyclerView recyclerView) {
     }
 
     private StreamSegmentAdapter.StreamSegmentListener getStreamSegmentListener() {
-        return (item, seconds) -> {
-            segmentAdapter.selectSegment(item);
-            player.seekTo(seconds * 1000L);
-            player.triggerProgressUpdate();
+        return new StreamSegmentAdapter.StreamSegmentListener() {
+            @Override
+            public void onItemClick(@NonNull final StreamSegmentItem item, final int seconds) {
+                segmentAdapter.selectSegment(item);
+                player.seekTo(seconds * 1000L);
+                player.triggerProgressUpdate();
+            }
+
+            @Override
+            public void onItemLongClick(@NonNull final StreamSegmentItem item, final int seconds) {
+                final PlayQueueItem currentItem = player.getCurrentItem();
+                if (currentItem != null) {
+                    String videoUrl = player.getVideoUrl();
+                    videoUrl += ("&t=" + seconds);
+                    ShareUtils.shareText(context, currentItem.getTitle(),
+                            videoUrl, currentItem.getThumbnailUrl());
```

Comment:
This won't work for services other than YouTube. You should use the same logic as in the player: https://github.com/TeamNewPipe/NewPipe/blob/4a7af6f9ac50c0a5a3ed3b306d08d4b8aa622aea/app/src/main/java/org/schabi/newpipe/player/Player.java#L1782

Reply:
Hey sorry for clarification, do you mean that I should be surrounding line 714 with an if statement checking that the service ID is indeed YouTube? I did look at this part in the player but thought it might not be necessary since I thought that these click handlers on the segments would only ever be called from a YouTube service. Sorry if I am misunderstanding and thanks for the quick replies!
<sub>And yes I am an ANU student :)</sub>

Reply:
> I thought that these click handlers on the segments would only ever be called from a YouTube service

That's probably true, but who knows if a future service will support segments ;-)

Yes, sorround it with a "is youtube" check. If it's not YouTube, don't allow long-pressing the item at all.

Reply:
Hey, I have been attempting to add this check to the `buildSegements` function with `setLongClickable(false)`, however even with `setOnLongClickListener(null)`, the long click event still occurs. Similarly, the existing `onClick` method used for seeking to different chapters does not get disabled by setting clickable to false. I have also tried adding the check to the beginning of the long click call, however since the call consumes the click event, this is not the desired result.

> don't allow long-pressing the item at all.  

Is there some better way of doing this that I am unaware of? Thanks

Reply:
Ok, then it's fine if you just don't do anything when a long-press is detected but the url is non-youtube